packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -514,6 +548,35 @@ class Scaffold extends StatefulWidget {
     );
   }
 
+  /// Returns a [ValueNotifier] for the [ScaffoldGeometry] for the closest
+  /// [Scaffold] ancestor of the given context.
+  ///
+  /// Value notifications are sent during the layout pass.
+  static ValueNotifier<ScaffoldGeometry> geometryOf(BuildContext context) {
+    final _ScaffoldScope scaffoldScope = context.inheritFromWidgetOfExactType(_ScaffoldScope);
+    if (scaffoldScope == null)
+    throw new FlutterError(
+      'Scaffold.geometryOf() called with a context that does not contain a Scaffold.\n'
+      'This usually happens when the context provided is from the same StatefulWidget as that '
+      'whose build function actually creates the Scaffold widget being sought.\n'
+      'There are several ways to avoid this problem. The simplest is to use a Builder to get a '
+      'context that is "under" the Scaffold. For an example of this, please see the '
+      'documentation for Scaffold.of():\n'
+      '  https://docs.flutter.io/flutter/material/Scaffold/of.html\n'
+      'A more efficient solution is to split your build function into several widgets. This '
+      'introduces a new context from which you can obtain the Scaffold. In this solution, '
+      'you would have an outer widget that creates the Scaffold populated by instances of '
+      'your new inner widgets, and then in these inner widgets you would use Scaffold.geometryOf().\n'
+      'A less elegant but more expedient solution is assign a GlobalKey to the Scaffold, '
+      'then use the key.currentState property to obtain the ScaffoldState rather than '
+      'using the Scaffold.geometryOf() function.\n'
```

Comment:
this message doesn't make sense. You can't use a GlobalKey to get the geometry value notifier.

Reply:
oops, clumsy copy-pasting